src/classes/queue-events.ts

```
@@ -194,64 +199,77 @@ export class QueueEvents extends QueueBase {
+                const args = array2obj(events[i][1]);
 
-            //
-            // TODO: we may need to have a separate xtream for progress data
-            // to avoid this hack.
-            switch (args.event) {
-              case 'progress':
-                args.data = JSON.parse(args.data);
-                break;
-              case 'completed':
-                args.returnvalue = JSON.parse(args.returnvalue);
-                break;
-            }
+                //
+                // TODO: we may need to have a separate xtream for progress data
+                // to avoid this hack.
+                switch (args.event) {
+                  case 'progress':
+                    args.data = JSON.parse(args.data);
+                    break;
+                  case 'completed':
+                    args.returnvalue = JSON.parse(args.returnvalue);
+                    break;
+                }
 
-            const { event, ...restArgs } = args;
+                const { event, ...restArgs } = args;
 
-            if (event === 'drained') {
-              this.emit(event, id);
-            } else {
-              this.emit(event, restArgs, id);
-              this.emit(`${event}:${restArgs.jobId}`, restArgs, id);
+                if (event === 'drained') {
+                  this.emit(event, id);
+                } else {
+                  this.emit(event, restArgs, id);
+                  this.emit(`${event}:${restArgs.jobId}`, restArgs, id);
+                }
+              }
+            }
+          } catch (err) {
+            if ((err as Error).message !== CONNECTION_CLOSED_ERROR_MSG) {
+              throw err;
             }
+            await delay(5000);
           }
         }
-      } catch (err) {
-        if ((err as Error).message !== CONNECTION_CLOSED_ERROR_MSG) {
-          throw err;
-        }
-        await delay(5000);
+      } catch (error) {
+        this.running = false;
+        this.emit('error', error);
```

Instead of emitting an error I think we just should throw the error so that the caller can handle it.